clang/lib/CodeGen/CGCall.h

```diff
@@ -285,6 +285,13 @@ class CallArgList : public SmallVector<CallArg, 8> {
 
     /// A value to "use" after the writeback, or null.
     llvm::Value *ToUse;
+
+    /// An Expression representing a cast from the temporary's type to the
+    /// source l-value's type.
+    const Expr *CastExpr;
```

Comment:
This comment is no longer correct; consider something like `WritebackExpr`. And maybe the logic for writebacks should simply be that we evaluate this expression if it's present, rather than making a language-mode decision?